datastore/postgres/updatevulnerabilities.go

```diff
@@ -45,14 +45,33 @@ var (
 	)
 )
 
+type deltaOpts struct {
+	deletedVulns []string
+	vulns        []*claircore.Vulnerability
+}
```

Comment:
I dislike "options structs" as a style issue. I think a `deleted Iter[string]` would be fine in the "internal" signature. I can harass @crozzy into helping bash that into shape -- just pull the "delta" changes into a commit that we can mess with.

Reply:
@hdonnay:

> I dislike "options structs" as a style issue. 
What are the pain points, in your opinion?

> I think a `deleted Iter[string]` would be fine in the "internal" signature.
The internal method needs all vulns upfront when `delete=true`. It makes things a little less straightforward to change if I have to iterate over the vulns when `delete=true` to handle the deletion steps first (they come first in the update logic) to decide then not to iterate if `delete=false` and iterate if `delete=false`.

The proposed patch seems the most straightforward way to achieve what we want; Considering we are sunsetting this interface sometime in the future.

Reply:
Ah, missed the fact that the iterator for this particular call can be called twice. For the internal API I think this assumption is fair. I am going to update this.